src/librustc_trans/debuginfo/gdb.rs

```diff
@@ -22,19 +23,13 @@ use std::ptr;
 use syntax::attr;
 
 
-/// Inserts a side-effect free instruction sequence that makes sure that the
-/// .debug_gdb_scripts global is referenced, so it isn't removed by the linker.
-pub fn insert_reference_to_gdb_debug_scripts_section_global(ccx: &CrateContext, builder: &Builder) {
+/// Inserts the .debug_gdb_scripts global into the set of symbols
+/// to be placed in `llvm.used`, so it isn't removed by the linker.
```

Comment:
No, by the linker. I thought that too, and started writing a comment, but turns out that it's not true; LLVM has a separate `!llvm.compiler.used` for what you say, and `!llvm.used` is [defined](http://llvm.org/docs/LangRef.html#the-llvm-used-global-variable) to keep the symbol through the linking process as well.